src/Plugin/Field/FieldFormatter/StrawberryBaseFormatter.php

```diff
@@ -0,0 +1,233 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: mlongley
+ * Date: 9/25/19
+ * Time: 8:56 PM
+ */
+
+namespace Drupal\format_strawberryfield\Plugin\Field\FieldFormatter;
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\FormatterBase;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Entity\EntityInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\format_strawberryfield\Tools\IiifUrlValidator;
+use Drupal\Core\Access\AccessResult;
+
+
+
+abstract class StrawberryBaseFormatter extends FormatterBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The Config for getting default IIIF settings.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface;
+   */
+  protected $iiifConfig;
+
+  /**
+   * The Config for getting formatter settings based on the View Mode
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface;
+   */
+  protected $viewModeConfig;
+
+  /**
+   *
+   * Whether or not new global IIIF urls conflict with ones previously set in config on a new module install.
+   * Set here so can be accessed in the static defaultSettings function
+   *
+   * @var boolean;
+   */
+//  protected $globalsConflict;
+
+  /**
+   * @param string $plugin_id
+   * @param mixed $plugin_definition
+   * @param \Drupal\Core\Field\FieldDefinitionInterface $field_definition
+   *   The definition of the field to which the formatter is associated.
+   * @param array $settings
+   * @param string $label
+   *   The formatter settings.
+   * @param $view_mode
+   *   The view mode.
+   * @param array $third_party_settings
+   *   Any third party settings.
+   *   The definition of the field to which the formatter is associated.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   */
+  public function __construct(
+    $plugin_id,
+    $plugin_definition,
+    FieldDefinitionInterface $field_definition,
+    array $settings,
+    $label,
+    $view_mode,
+    array $third_party_settings,
+    ConfigFactoryInterface $config_factory
+  ) {
+    parent::__construct(
+      $plugin_id,
+      $plugin_definition,
+      $field_definition,
+      $settings,
+      $label,
+      $view_mode,
+      $third_party_settings
+    );
+    $this->iiifConfig = $config_factory->get('format_strawberryfield.iiif_settings');
+    $this->viewModeConfig =  $config_factory->getEditable('core.entity_view_display.node.digital_object.'.$view_mode);
```

Comment:
$this->viewModeConfig is not needed anymore, so can be removed from the constructor and as a class property `protected $viewModeConfig;`